hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerImpl.java

```diff
@@ -219,25 +274,24 @@ public Pipeline createPipeline(ReplicationConfig replicationConfig,
       LOG.info(message);
       throw new IOException(message);
     }
+  }
 
+  private void addPipelineToManager(Pipeline pipeline)
+      throws IOException, TimeoutException {
     acquireWriteLock();
-    final Pipeline pipeline;
     try {
-      pipeline = pipelineFactory.create(replicationConfig,
-          excludedNodes, favoredNodes);
       stateManager.addPipeline(pipeline.getProtobufMessage(
           ClientVersion.CURRENT_VERSION));
     } catch (IOException | TimeoutException ex) {
-      LOG.debug("Failed to create pipeline with replicationConfig {}.",
-          replicationConfig, ex);
+      LOG.debug("Failed to add pipeline with replicationConfig {}.",
+          pipeline.getReplicationConfig(), ex);
```

Comment:
Nit: we may want to log the pipeline itself.  Previously this was called with a fresh pipeline, but now `pipeline` can be "anything".